### ISSUE TITLE
2019.4: test suite is failing

### ISSUE THREAD

**Issue by kloczek** [KhronosGroup/SPIRV-Tools#3088]
<pre>+ /usr/bin/ninja test -v -j48
[0/1] cd /home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-3e4abc9ac382b6fc7a88316a700652404b6fb9f7 &amp;&amp; /usr/bin/ctest --force-new-ctest-process
Test project /home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-3e4abc9ac382b6fc7a88316a700652404b6fb9f7
    Start 1: spirv-tools-copyrights
1/9 Test #1: spirv-tools-copyrights ..........................***Failed    0.26 sec
    Start 2: spirv-tools-symbol-exports-SPIRV-Tools
2/9 Test #2: spirv-tools-symbol-exports-SPIRV-Tools ..........***Failed    0.09 sec
    Start 3: spirv-tools-symbol-exports-SPIRV-Tools-shared
3/9 Test #3: spirv-tools-symbol-exports-SPIRV-Tools-shared ...***Failed    0.09 sec
    Start 4: spirv-tools-symbol-exports-SPIRV-Tools-opt
4/9 Test #4: spirv-tools-symbol-exports-SPIRV-Tools-opt ......   Passed    0.14 sec
    Start 5: spirv-tools-symbol-exports-SPIRV-Tools-reduce
5/9 Test #5: spirv-tools-symbol-exports-SPIRV-Tools-reduce ...   Passed    0.06 sec
    Start 6: spirv-tools-symbol-exports-SPIRV-Tools-link
6/9 Test #6: spirv-tools-symbol-exports-SPIRV-Tools-link .....   Passed    0.05 sec
    Start 7: expect_nosetests
7/9 Test #7: expect_nosetests ................................***Failed    0.25 sec
    Start 8: spirv_test_framework_nosetests
8/9 Test #8: spirv_test_framework_nosetests ..................***Failed    0.24 sec
    Start 9: spirv_opt_cli_tools_tests
9/9 Test #9: spirv_opt_cli_tools_tests .......................   Passed    0.36 sec

44% tests passed, <font color="#CC0000">5 tests failed</font> out of 9

Total Test time (real) =   1.56 sec

The following tests FAILED:
	<font color="#CC0000">  1 - spirv-tools-copyrights (Failed)</font>
	<font color="#CC0000">  2 - spirv-tools-symbol-exports-SPIRV-Tools (Failed)</font>
	<font color="#CC0000">  3 - spirv-tools-symbol-exports-SPIRV-Tools-shared (Failed)</font>
	<font color="#CC0000">  7 - expect_nosetests (Failed)</font>
	<font color="#CC0000">  8 - spirv_test_framework_nosetests (Failed)</font>
Errors while running CTest
FAILED: CMakeFiles/test.util 
cd /home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-3e4abc9ac382b6fc7a88316a700652404b6fb9f7 &amp;&amp; /usr/bin/ctest --force-new-ctest-process
</pre>

**Comment by s-perron** [comment]
@zoddicus Do you know what needs to be done here?  Is the fix to pick the right dependencies?  Or is it a python issue?

**Comment by dj2** [comment]
Can you run the tests individually so we can see what caused the failure? (or alternatively, run ctest with CTEST_OUTPUT_ON_FAILURE=1  environment variable set).

**Comment by zoddicus** [comment]
This smells like python versioning issues, since the nosetests are failing. logs like dj2 suggested would be helpful in confirming this.

The hash in the path name, which I assume is the version you are trying to build, is about a month old. Is there a need to be building this specific version of the library?

I know I have landed Python fixes since that hash. Specifically I actually removed the nosetests recently because there was no simple way to make them Python3 compatible. They have been replaced with unittest based tests.

**Comment by kloczek** [comment]
Indeed it was wrong/old version of the spirv-headers.
It would be really nice to add to spirv-headers pkgconfig file and in spirv-tools detection version ot he spirv-headers to avoid that kind of errors and display error message that system needs to have enough fresh spirv-headers package.

Feel free to close that ticket but I would keep opened it until someone will not add pkgconfig files and checks :)

**Comment by kloczek** [comment]
Sorry I've mixed two things with not building .spirv-tools because to old spirv-headers.

Just have been working on refresh my package with spirv-tools from 1.5.1 to latest 1.5.3. I've upgraded spirv-headers. After that spirv-tools build correctly 
Test suite still is failing.
I;m using at the moment python 3.8.3.
```
[tkloczko@barrel SPIRV-Tools-2020.3]$ make check
make: *** No rule to make target 'check'.  Stop.
[tkloczko@barrel SPIRV-Tools-2020.3]$ make test
Running tests...
/usr/bin/ctest --force-new-ctest-process
Test project /home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3
    Start 1: spirv-tools-copyrights
1/9 Test #1: spirv-tools-copyrights ..........................***Failed    0.36 sec
    Start 2: spirv-tools-symbol-exports-SPIRV-Tools
2/9 Test #2: spirv-tools-symbol-exports-SPIRV-Tools ..........***Failed    0.12 sec
    Start 3: spirv-tools-symbol-exports-SPIRV-Tools-shared
3/9 Test #3: spirv-tools-symbol-exports-SPIRV-Tools-shared ...***Failed    0.12 sec
    Start 4: spirv-tools-symbol-exports-SPIRV-Tools-opt
4/9 Test #4: spirv-tools-symbol-exports-SPIRV-Tools-opt ......   Passed    0.20 sec
    Start 5: spirv-tools-symbol-exports-SPIRV-Tools-reduce
5/9 Test #5: spirv-tools-symbol-exports-SPIRV-Tools-reduce ...   Passed    0.10 sec
    Start 6: spirv-tools-symbol-exports-SPIRV-Tools-link
6/9 Test #6: spirv-tools-symbol-exports-SPIRV-Tools-link .....   Passed    0.08 sec
    Start 7: spirv-tools_expect_unittests
7/9 Test #7: spirv-tools_expect_unittests ....................   Passed    0.12 sec
    Start 8: spirv-tools_spirv_test_framework_unittests
8/9 Test #8: spirv-tools_spirv_test_framework_unittests ......   Passed    0.11 sec
    Start 9: spirv_opt_cli_tools_tests
9/9 Test #9: spirv_opt_cli_tools_tests .......................   Passed    0.45 sec

67% tests passed, 3 tests failed out of 9

Total Test time (real) =   1.68 sec

The following tests FAILED:
          1 - spirv-tools-copyrights (Failed)
          2 - spirv-tools-symbol-exports-SPIRV-Tools (Failed)
          3 - spirv-tools-symbol-exports-SPIRV-Tools-shared (Failed)
Errors while running CTest
make: *** [Makefile:141: test] Error 8
```
```
[tkloczko@barrel SPIRV-Tools-2020.3]$ cmake -VV
CMake Error: The source directory "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/-VV" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
[tkloczko@barrel SPIRV-Tools-2020.3]$ ctest -VV
UpdateCTestConfiguration  from :/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/DartConfiguration.tcl
Test project /home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
    Start 1: spirv-tools-copyrights

1: Test command: /usr/bin/python3 "utils/check_copyright.py"
1: Test timeout computed to be: 10000000
1: utils/check_copyright.py:130: SyntaxWarning: "is" with a literal. Did you mean "=="?
1:   if state is 0:
1: utils/check_copyright.py:141: SyntaxWarning: "is" with a literal. Did you mean "=="?
1:   elif state is 1:
1: utils/check_copyright.py:149: SyntaxWarning: "is" with a literal. Did you mean "=="?
1:   elif state is 2:
1: ./OpenCLDebugInfo100.h has no Apache 2 license notice
1: ./DebugInfo.h has no Apache 2 license notice
1/9 Test #1: spirv-tools-copyrights ..........................***Failed    0.35 sec
test 2
    Start 2: spirv-tools-symbol-exports-SPIRV-Tools

2: Test command: /usr/bin/python3 "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/utils/check_symbol_exports.py" "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/source/libSPIRV-Tools.so"
2: Test timeout computed to be: 10000000
2: check_symbol_exports: error: Unescaped exported symbol: _ZSt3decRSt8ios_base
2: check_symbol_exports: error: Unescaped exported symbol: _ZSt3hexRSt8ios_base
2/9 Test #2: spirv-tools-symbol-exports-SPIRV-Tools ..........***Failed    0.12 sec
test 3
    Start 3: spirv-tools-symbol-exports-SPIRV-Tools-shared

3: Test command: /usr/bin/python3 "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/utils/check_symbol_exports.py" "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/source/libSPIRV-Tools-shared.so"
3: Test timeout computed to be: 10000000
3: check_symbol_exports: error: Unescaped exported symbol: _ZSt3decRSt8ios_base
3: check_symbol_exports: error: Unescaped exported symbol: _ZSt3hexRSt8ios_base
3/9 Test #3: spirv-tools-symbol-exports-SPIRV-Tools-shared ...***Failed    0.12 sec
test 4
    Start 4: spirv-tools-symbol-exports-SPIRV-Tools-opt

4: Test command: /usr/bin/python3 "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/utils/check_symbol_exports.py" "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/source/opt/libSPIRV-Tools-opt.so"
4: Test timeout computed to be: 10000000
4/9 Test #4: spirv-tools-symbol-exports-SPIRV-Tools-opt ......   Passed    0.20 sec
test 5
    Start 5: spirv-tools-symbol-exports-SPIRV-Tools-reduce

5: Test command: /usr/bin/python3 "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/utils/check_symbol_exports.py" "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/source/reduce/libSPIRV-Tools-reduce.so"
5: Test timeout computed to be: 10000000
5/9 Test #5: spirv-tools-symbol-exports-SPIRV-Tools-reduce ...   Passed    0.10 sec
test 6
    Start 6: spirv-tools-symbol-exports-SPIRV-Tools-link

6: Test command: /usr/bin/python3 "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/utils/check_symbol_exports.py" "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/source/link/libSPIRV-Tools-link.so"
6: Test timeout computed to be: 10000000
6/9 Test #6: spirv-tools-symbol-exports-SPIRV-Tools-link .....   Passed    0.07 sec
test 7
    Start 7: spirv-tools_expect_unittests

7: Test command: /usr/bin/python3 "-m" "unittest" "expect_unittest.py"
7: Test timeout computed to be: 10000000
7: ....
7: ----------------------------------------------------------------------
7: Ran 4 tests in 0.000s
7:
7: OK
7/9 Test #7: spirv-tools_expect_unittests ....................   Passed    0.10 sec
test 8
    Start 8: spirv-tools_spirv_test_framework_unittests

8: Test command: /usr/bin/python3 "-m" "unittest" "spirv_test_framework_unittest.py"
8: Test timeout computed to be: 10000000
8: ..
8: ----------------------------------------------------------------------
8: Ran 2 tests in 0.001s
8:
8: OK
8/9 Test #8: spirv-tools_spirv_test_framework_unittests ......   Passed    0.10 sec
test 9
    Start 9: spirv_opt_cli_tools_tests

9: Test command: /usr/bin/python3 "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/test/tools/opt/../spirv_test_framework.py" "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/tools/spirv-opt" "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/tools/spirv-as" "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/tools/spirv-dis" "--test-dir" "/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/test/tools/opt"
9: Test timeout computed to be: 10000000
9: /home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/test/tools/opt/../spirv_test_framework.py:149: SyntaxWarning: "is" with a literal. Did you mean "=="?
9:   if sys.version_info[0] is 2:
9: /home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/test/tools/opt/../spirv_test_framework.py:151: SyntaxWarning: "is" with a literal. Did you mean "=="?
9:   elif sys.version_info[0] is 3:
9: /home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/test/tools/opt/../spirv_test_framework.py:159: SyntaxWarning: "is" with a literal. Did you mean "=="?
9:   if sys.version_info[0] is 2:
9: /home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.3/test/tools/opt/../spirv_test_framework.py:161: SyntaxWarning: "is" with a literal. Did you mean "=="?
9:   elif sys.version_info[0] is 3:
9: SPIRV tool test suite: "SpirvOptBase"
9: 1/24       TestAssemblyFileAsOnlyParameter          Passed
9: SPIRV tool test suite: "SpirvOptFlags"
9: 2/24       TestHelpFlag                             Passed
9: 3/24       TestLegalizationPasses                   Passed
9: 4/24       TestLoopFissionArgsInvalidNumber         Passed
9: 5/24       TestLoopFissionArgsNegative              Passed
9: 6/24       TestLoopFusionArgsInvalidNumber          Passed
9: 7/24       TestLoopFusionArgsNegative               Passed
9: 8/24       TestLoopPeelingThresholdArgsInvalidNumber Passed
9: 9/24       TestLoopPeelingThresholdArgsNegative     Passed
9: 10/24      TestLoopUnrollPartialArgsInvalidNumber   Passed
9: 11/24      TestLoopUnrollPartialArgsNegative        Passed
9: 12/24      TestPerformanceOptimizationPasses        Passed
9: 13/24      TestScalarReplacementArgsInvalidNumber   Passed
9: 14/24      TestScalarReplacementArgsNegative        Passed
9: 15/24      TestSizeOptimizationPasses               Passed
9: 16/24      TestTargetEnvThenVulkanToWebGPUIsInvalid Passed
9: 17/24      TestTargetEnvThenWebGPUToVulkanIsInvalid Passed
9: 18/24      TestValidPassFlags                       Passed
9: 19/24      TestVulkanToWebGPUThenTargetEnvIsInvalid Passed
9: 20/24      TestVulkanToWebGPUThenWebGPUToVulkanIsInvalid Passed
9: 21/24      TestWebGPUToVulkanThenTargetEnvIsInvalid Passed
9: 22/24      TestWebGPUToVulkanThenVulkanToWebGPUIsInvalid Passed
9: SPIRV tool test suite: "SpirvOptConfigFile"
9: 23/24      TestOconfigComments                      Passed
9: 24/24      TestOconfigEmpty                         Passed
9/9 Test #9: spirv_opt_cli_tools_tests .......................   Passed    0.42 sec

67% tests passed, 3 tests failed out of 9

Total Test time (real) =   1.62 sec

The following tests FAILED:
          1 - spirv-tools-copyrights (Failed)
          2 - spirv-tools-symbol-exports-SPIRV-Tools (Failed)
          3 - spirv-tools-symbol-exports-SPIRV-Tools-shared (Failed)
Errors while running CTest
```

**Comment by kloczek** [comment]
Just tested 3030.5 and looks like still some units are failing
```
+ /usr/bin/make -O -j48 V=1 VERBOSE=1 -C x86_64-redhat-linux-gnu test ARGS=--output-on-failure
make: Entering directory '/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.5/x86_64-redhat-linux-gnu'
Running tests...
/usr/bin/ctest --force-new-ctest-process --output-on-failure
Test project /home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.5/x86_64-redhat-linux-gnu
    Start 1: spirv-tools-copyrights
1/9 Test #1: spirv-tools-copyrights ..........................***Failed    0.37 sec
utils/check_copyright.py:131: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if state is 0:
utils/check_copyright.py:142: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif state is 1:
utils/check_copyright.py:150: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif state is 2:
./x86_64-redhat-linux-gnu/DebugInfo.h has no Apache 2 license notice
./x86_64-redhat-linux-gnu/OpenCLDebugInfo100.h has no Apache 2 license notice

    Start 2: spirv-tools-symbol-exports-SPIRV-Tools-static
2/9 Test #2: spirv-tools-symbol-exports-SPIRV-Tools-static ...   Passed    0.12 sec
    Start 3: spirv-tools-symbol-exports-SPIRV-Tools-shared
3/9 Test #3: spirv-tools-symbol-exports-SPIRV-Tools-shared ...***Failed    0.09 sec
check_symbol_exports: error: Unescaped exported symbol: _ZSt3decRSt8ios_base
check_symbol_exports: error: Unescaped exported symbol: _ZSt3hexRSt8ios_base

    Start 4: spirv-tools-symbol-exports-SPIRV-Tools-opt
4/9 Test #4: spirv-tools-symbol-exports-SPIRV-Tools-opt ......   Passed    0.26 sec
    Start 5: spirv-tools-symbol-exports-SPIRV-Tools-reduce
5/9 Test #5: spirv-tools-symbol-exports-SPIRV-Tools-reduce ...   Passed    0.10 sec
    Start 6: spirv-tools-symbol-exports-SPIRV-Tools-link
6/9 Test #6: spirv-tools-symbol-exports-SPIRV-Tools-link .....   Passed    0.07 sec
    Start 7: spirv-tools_expect_unittests
7/9 Test #7: spirv-tools_expect_unittests ....................   Passed    0.10 sec
    Start 8: spirv-tools_spirv_test_framework_unittests
8/9 Test #8: spirv-tools_spirv_test_framework_unittests ......   Passed    0.09 sec
    Start 9: spirv_opt_cli_tools_tests
9/9 Test #9: spirv_opt_cli_tools_tests .......................   Passed    0.27 sec

78% tests passed, 2 tests failed out of 9

Total Test time (real) =   1.49 sec

The following tests FAILED:
          1 - spirv-tools-copyrights (Failed)
          3 - spirv-tools-symbol-exports-SPIRV-Tools-shared (Failed)
Errors while running CTest
make: *** [Makefile:140: test] Error 8
make: Leaving directory '/home/tkloczko/rpmbuild/BUILD/SPIRV-Tools-2020.5/x86_64-redhat-linux-gnu'
```
Symbols isues maybe caused by using in this case LTO.

**Comment by dj2** [comment]
For the copyright checks, what version of python are you running? I think they expect python3?

The symbol error looks like it's coming from the stdlib. `std::hex(std::ios_base&)` can you turn off LTO and see if it builds?

**Comment by kloczek** [comment]
I think that thi ticket can be closed now as Moved to 2021.1 https://github.com/KhronosGroup/SPIRV-Tools/issues/4250
